### DISCUSSION TITLE
Component development

### DISCUSSION THREAD

**Discussion by vnermolaev** [DioxusLabs/dioxus#2471] [General]
Hi everyone,

I have developed a UI component using Dioxus, and I am looking to open-source it as a crate. The component includes custom CSS, assets managed by Manganis, and utilizes Tailwind CSS for styling. Unfortunately, I haven't been able to find clear instructions or guidelines on how to package and publish a Dioxus component with these dependencies.

I did find a library with Dioxus components [here](https://github.com/DioxusLabs/example), but it appears to be unmaintained, and it doesn't cover the specifics of managing custom CSS, assets, and Tailwind CSS.

Here are some specific questions I have:

- **Bundling CSS and Tailwind**: How should I include my custom CSS and Tailwind CSS configuration in the crate? Should I precompile the CSS, or is there a preferred method to ensure the styles are correctly applied when someone uses the crate?
- **Managing Assets with Manganis**: What is the best practice for including and managing assets (like images or fonts) within the crate? How should these assets be bundled and referenced?
- **Publishing the Crate**: Are there any special considerations or steps I need to follow to publish a Dioxus component crate, particularly with these additional dependencies?

Any guidance or examples would be greatly appreciated. Thank you!

**Comment by ealmloff** [comment]
If you are using manganis to include styles and assets, everything should work automatically when you publish it just like a normal crate. The dioxus CLI looks in dependencies for any assets they use automatically

**Comment by vnermolaev** [comment]
Hey, thanks for the reply.

I was able to include all the custom assets and use them in a dependent crate. However, I struggled to integrate Tailwind styles using `manganis::classes!`. Instead, I compiled the necessary tailwind classes myself and included them as a custom asset.
